lint.py

```diff
@@ -37,39 +37,81 @@ def _run_linter_one(args, linter, root, filename, validate_only):
                git_tools.COLORS.NORMAL, lex.message),
               file=sys.stderr)
         return filename, lex.fixable
+
     if contents != new_contents:
-        violations_message = ', '.join(
-            '%s%s%s' %
-            (git_tools.COLORS.FAIL, violation, git_tools.COLORS.NORMAL)
-            for violation in violations)
-        if validate_only:
-            print('File %s%s%s lint failed: %s' %
-                  (git_tools.COLORS.HEADER, filename,
-                   git_tools.COLORS.NORMAL, violations_message),
-                  file=sys.stderr)
-        else:
-            print('Fixing %s%s%s' %
-                  (git_tools.COLORS.HEADER, filename,
-                   git_tools.COLORS.NORMAL),
-                  file=sys.stderr)
-            with open(os.path.join(root, filename), 'wb') as outfile:
-                outfile.write(new_contents)
-        return filename, True
+      return _run_linter_mixed(
+        args, linter, root, filename, new_contents, validate_only, violations)
     return None, False
 
 
-def _run_linter(args, linter, files, validate_only):
+def _run_linter_all(args, linter, root, files, validate_only):
+    try:
+      new_file_contents, original_contents, violations = linter.run_all(files,
+        lambda filename: git_tools.file_contents(args, root, filename))
+    except linters.LinterException as lex:
+        print('Files %s%s%s lint failed:\n%s' %
+              (git_tools.COLORS.FAIL, files,
+               git_tools.COLORS.NORMAL, lex.message),
+              file=sys.stderr)
+        return [(files[0], lex.fixable)]
+
+    result = []
+    for filename in new_file_contents:
+      contents = (new_file_contents[filename] if filename in new_file_contents
```

Comment:
aquí `filename in new_file_contents` está garantizado a ser verdadero porque estás iterando ese mapa. mejor en vez de hacer eso haz

```python
for filename, contents in new_file_contents:
```

Reply:
listo